File: scripts/xmlrpc_console.rb
require 'XMLRPC/Client'

server = XMLRPC::Client.new2('http://mac-mini.local/~petevilter/svn/braindump/trunk/xmlrpc.php')

while 1
  $stdout.print 'braindump> '
  query = $stdin.gets
  puts server.call('braindump.query',query)
end

# TODO:
# humanize result arrays
# say something if result is 'false'
# 'set' and 'list' don't work

File: scripts/xmlrpc_console.py
import xmlrpclib, sys

server = xmlrpclib.Server('http://mac-mini.home/~petevilter/svn/braindump/trunk/xmlrpc.php')

try:  
  while 1:
    query = raw_input('braindump> ')
    print server.braindump.query(query)
    
except KeyboardInterrupt: # ctrl-c
  print "\nbye!"
  sys.exit(0)
  
# TODO:
# humanize result arrays
# give an english message instead of 'false' or 'true'

# FIXME:
# changing attributes doesn't work cuz it doesn't unset the old value